kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/data/DataServiceImpl.java

```diff
@@ -989,7 +989,6 @@ public void startConnectionTask() {
 
     @Override
     public void stopConnectionTask() {
-        disconnect();
```

Comment:
@nicolatimeus you added this line in #4023 . Can we remove it safetely ?

Reply:
I don't remember the reason for this. We can try removing it and see if it causes any regression.

Reply:
It seems to be okay with it removed. It appears everything gets shutdown as expected. Will ask QA for some support.